src/core/openthread.cpp

```diff
@@ -1256,9 +1256,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
     ThreadError error;
+    Message *message;
+    Ip6::Header header;
 
     otLogFuncEntry();
 
+    message = static_cast<Message *>(aMessage);
+    assert(message->GetLength() >= sizeof(header));
+
+    message->Read(0, sizeof(header), &header);
+    assert(!header.GetDestination().IsRealmLocalMulticast() || header.GetNextHeader() != Ip6::kProtoHopOpts);
```

Comment:
@LuDuda, the scenario for removing the MPL Option is when OpenThread passes a received IPv6 datagram to the host via `otReceiveIp6DatagramCallback`.  @nibanks asked the question of whether we should remove the MPL Option so that the host does not need to know that MPL is even happening.